File: LeetCode1073/LeetCode1073.py
class Solution:
	def addNegabinary(self, arr1: List[int], arr2: List[int]) -> List[int]:
		result = []
		length1 = len(arr1) - 1
		length2 = len(arr2) - 1
		length = length1 if length1 > length2 else length2

		carry = 0

		while length >= 0 or carry != 0:
			term1 = arr1[length1] if length1 >= 0 else 0
			term2 = arr2[length2] if length2 >= 0 else 0

			sum = term1 + term2 + carry
			carry = 0

			if sum > 1:
				sum -= 2
				carry = -1
			elif sum < 0:
				sum += 2
				carry = 1

			result.append(sum)
			length1 -= 1
			length2 -= 1
			length -= 1

		result.reverse()

		while (result[0] == 0 and len(result) > 1):
			result.pop()

		return result

File: LeetCode1073/LeetCode1073.cpp
#include <vector>

class Solution {
public:
    vector<int> addNegabinary(std::vector<int>& arr1, std::vector<int>& arr2) {
        std::vector<int> result;
        
        int i = arr1.size() - 1;
        int j = arr2.size() - 1;
        int length = i > j ? i: j;
        int carry = 0;

        while (length >= 0 || carry != 0)
        {
            int term1 = i >= 0 ? arr1[i] : 0;
            int term2 = j >= 0 ? arr2[j] : 0;

            int sum = term1 + term2 + carry;
            carry = 0;
            
            if (sum > 1)
            {
                sum -= 2;
                carry = -1;
            }
            else if (sum < 0)
            {
                sum += 2;
                carry = 1;
            }

            result.push_back(sum);
            i--;
            j--;
            length--;
        }

        std::reverse(result.begin(), result.end());
        
        while (result[0] == 0 && result.size() > 1)
        {
            result.erase(result.begin());
        }

        return result;
    }
};